pilot/pkg/simulation/traffic.go

```diff
@@ -427,7 +427,7 @@ func (sim *Simulation) matchFilterChain(chains []*listener.FilterChain, defaultC
 	}, func(fc *listener.FilterChainMatch) bool {
 		if !hasTLSInspector {
 			// Without tls inspector, transport protocol will always be raw buffer
-			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol
+			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol && input.TLS != MTLS
```

Comment:
Not sure about this change. If we don't have tls inspector defined, how would mtls traffic match `raw_buffer`?

Reply:
Actually shouldn't we return false to indicate an error if there is no tls_inspector but the input.TLS is NOT plain text?

I changed it to `input.TLS == PLAINTEXT` and updated the test `TestInbound/passthrough_tls` as the pass through filter chain should reject the TLS request if the global peer authN is disable.